lib/ansible/modules/source_control/git.py

```diff
@@ -1023,13 +1058,25 @@ def main():
     # call run_command()
     module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C', LC_CTYPE='C')
 
+    if separate_git_dir is not None and bare:
+        module.warn("Ignoring separate_git_dir argument. "
+                    "Can not do bare clone with argument separate_git_dir.")
+        separate_git_dir = None
+    elif separate_git_dir:
+        separate_git_dir = os.path.abspath(separate_git_dir)
+
     gitconfig = None
     if not dest and allow_clone:
         module.fail_json(msg="the destination directory must be specified unless clone=no")
     elif dest:
         dest = os.path.abspath(dest)
         try:
             repo_path = get_repo_path(dest, bare)
+            if separate_git_dir and os.path.exists(repo_path) and separate_git_dir != repo_path:
```

Comment:
What if it was also `bare` and the user is changing it into non-bare along with setting up `separate_git_dir`?

Reply:
I go through some doc about `git`, it seems people will not frequently change a bare repo to a full repo. And to do that, the command is not related with `bare` or `separate_git_dir`, so I think we don't need to worry about this.